src/main/java/decodex/parser/Parser.java

```diff
@@ -57,27 +57,29 @@ public String getCommandType(String userInput) throws ParserException {
         return commandType;
     }
 
-    /**
-     * Returns the Argument portion of the user input as a whole string.
-     *
-     * @param userInput The input specified by the user.
-     * @return The argument portion of the user input as an array.
-     */
-    public String getUserArgument(String userInput) throws ParserException {
+    public String getModuleName(String userInput) throws ParserException {
         String strippedUserInput = userInput.stripLeading();
         String[] tokens = strippedUserInput.split(" ", -1);
 
-        if (tokens.length < VALID_TOKENS_LENGTH_FOR_ARGUMENTS) {
-            throw new ParserException(ParserException.MISSING_COMMAND_ARGS_MESSAGE);
+        String[] argumentTokens = Arrays.stream(Arrays.copyOfRange(tokens, STARTING_ARGUMENTS_INDEX, tokens.length))
+                .filter(value -> !value.isBlank())
+                .toArray(size -> new String[size]);
+        if (argumentTokens.length == 0) {
+            throw new ParserException("[-] Module name is missing");
         }
-        String[] argumentTokens = Arrays.copyOfRange(tokens, STARTING_ARGUMENTS_INDEX, tokens.length);
+        return argumentTokens[MODULE_NAME_INDEX_IN_TOKENS];
+    }
+
+    public String getInputString(String userInput) throws ParserException {
+        String strippedUserInput = userInput.stripLeading();
+        String[] tokens = strippedUserInput.split(" ", -1);
 
-        if (argumentTokens.length > 1) {
-            String joinedArguments = String.join(" ", argumentTokens);
-            return joinedArguments;
+        String argumentString = String.join(" ",
+                Arrays.copyOfRange(tokens, STARTING_ARGUMENTS_INDEX, tokens.length));
+        if (argumentString.isEmpty()) {
+            throw new ParserException("[-] Your input is empty, please do \"input i am a string\"");

```

Comment:
As this method is generally used for those that have an input string, perhaps change the message to something more general like `[-] Input is empty`?

Reply:
Agree